### ISSUE TITLE
Spanish model using spaCy (Rasa NLU)

### ISSUE THREAD

**Issue by parvathysarat** [explosion/spaCy#1477]
I am using Rasa NLU with spacy/sklearn to train a Spanish model.  My issue is that I am not sure if the spacy Spanish language model ('es_core_web_sm') has been loaded & my training entities are added to the already existing values, OR if the custom-trained texts/entities/values are being added to a blank model, because I'm not getting intents with good confidence levels for basic spanish sentences.

My config file (config_spacynew.json) :
`{ "language" : "es", "spacy_model_name" : "es", "pipeline": "spacy_sklearn", "path" : "./projects", "data" : "./stuff.json" }`

```

> 
> import spacy
> import requests
> import json
> import sklearn
> from rasa_nlu.converters import load_data
> from rasa_nlu.config import RasaNLUConfig
> from rasa_nlu.model import Trainer
> 
> training_data = load_data('stuff.json')
> trainer = Trainer(RasaNLUConfig("config_spacynew.json"))
> trainer.train(training_data)
> model_directory = trainer.persist('./projects/default/')
> from rasa_nlu.model import Metadata, Interpreter
> 
> # where `model_directory points to the folder the model is persisted in
> interpreter = Interpreter.load(model_directory, RasaNLUConfig("stuff.json"))
> print interpreter.parse(u"como estas")
> 
> Fitting 2 folds for each of 6 candidates, totalling 12 fits
> [Parallel(n_jobs=1)]: Done  12 out of  12 | elapsed:    0.1s finished
> {u'entities': [], u'intent': {u'confidence': 0.51586523913578286, u'name': u'greet'}, 'text': u'como estas', u'intent_ranking': [{u'confidence': 0.51586523913578286, u'name': u'greet'}, {u'confidence': 0.19394466990205211, u'name': u'hazard reporting'}, {u'confidence': 0.11179921846426252, u'name': u'affirm'}, {u'confidence': 0.061330332434286455, u'name': u'hazard logging'}, {u'confidence': 0.059356678259903005, u'name': u'safety recommendation'}, {u'confidence': 0.025392835701401759, u'name': u'situational awareness'}, {u'confidence': 0.017844732793248858, u'name': u'restaurant_search'}, {u'confidence': 0.014466293309062768, u'name': u'goodbye'}]}
> [Finished in 17.3s]
> 
```
'como estas' was part of my training set as well yet comes up with a really low confidence. So did 'hola' , which led me to believe the model is trained on my data alone. Also does having more examples for a particular intent skew the model to give that particular intent for any unknown cases? 
I read that training examples in scale of 100s or 1000s is considered 'decent' for spacy, how far is that true, even for intents such as greet? 

## Your Environment

* Operating System: 
* Python Version Used:  2.7
* spaCy Version Used: 1.9.0
* Environment Information:  Xubuntu

**Comment by ines** [comment]
Hmm, a lot of this code looks very specific to Rasa, so I'm afraid we won't really be able to help you with that. For questions around their library, the [Rasa issue tracker](https://github.com/RasaHQ/rasa_nlu/issues) might be a better place to ask.

However – good news! 🎉 We've just released a new alpha version of spaCy, `v2.0.0a18`, which comes with two Spanish models that both support tagging, parsing, NER and similarity: https://alpha.spacy.io/models/es

We've also updated the training examples and they're now much easier to run and adapt, and include more documentation. For example:

* [Training an intent parser with spaCy](https://github.com/explosion/spaCy/blob/develop/examples/training/train_intent_parser.py) – this might actually be very relevant to you. In this example, we're simply training spaCy's dependency parser to predict intent dependencies instead of syntactic dependencies. Sounds almost *too* simple, but works surprisingly well (I'm actually super excited about this example – there's so much cool stuff that can be done with this approach!). [See the updated docs](https://alpha.spacy.io/usage/training#intent-parser) for more details and a visualization.
* [Training a text classification model](https://github.com/explosion/spaCy/blob/develop/examples/training/train_textcat.py) – you'll need a larger, labelled dataset for this one, but you'd be able to teach spaCy to label entire documents as "about topic XY". This example trains on the IMDB movie reviews dataset.
* [Updating the named entity recognizer](https://github.com/explosion/spaCy/blob/develop/examples/training/train_ner.py) – probably necessary if the out-of-the-box predictions are not good enough on your data.
* [Training a new entity type](https://github.com/explosion/spaCy/blob/develop/examples/training/train_new_entity_type.py) – probably also relevant, if your application needs to recognise non-standard entities?

So I'd recommend trying out the above examples with the Spanish model and see how far you get and how good the results are. You might be able to pre-train a pretty decent model using only spaCy and then fine-tune the chatbot specifics with Rasa, once they support v2.0.

**Comment by parvathysarat** [comment]
Yes, a lot of the code is specific to Rasa, sorry about that. Wanted to know if this is a spaCy specific issue.

I have found that spaCy requires a bigger dataset to come up with decent confidence levels, though it's super fast in training. I'll keep adding intent/entity examples to see if and when I can crack it. Meanwhile, I've found MITIE to work on for my smaller dataset, though it takes (much) longer than spaCy to train.

Thanks for your time and inputs, Ines! Will take a look at v2.0 asap!
